packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -840,6 +844,157 @@ class ThemeData with Diagnosticable {
+            )
+            : ThemeData.from(
+              useMaterial3: useMaterial3,
+              colorScheme: colorScheme,
+              textTheme: textTheme,
+            );
+
+    if (systemColors == null) {
+      return theme;
+    }
+
+    return theme.copyWith(
+      elevatedButtonTheme: ElevatedButtonThemeData(
+        style: ElevatedButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      textButtonTheme: TextButtonThemeData(
+        style: TextButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      outlinedButtonTheme: OutlinedButtonThemeData(
+        style: OutlinedButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      filledButtonTheme: FilledButtonThemeData(
+        style: FilledButton.styleFrom(
+          foregroundColor: systemColors.buttonText.value,
+          backgroundColor: systemColors.buttonFace.value,
+          side:
+              systemColors.buttonBorder.value == null
+                  ? null
+                  : BorderSide(color: systemColors.buttonBorder.value!),
+        ),
+      ),
+      floatingActionButtonTheme: FloatingActionButtonThemeData(
+        backgroundColor: systemColors.buttonFace.value,
+        foregroundColor: systemColors.buttonText.value,
+      ),
+    );
+  }
+
+  /// Create a high contrast light theme based on a combination of:
+  ///
+  /// - A purple primary color that matches the baseline [Material 2](https://material.io/design/color/the-color-system.html#color-theme-creation)
+  ///   or [Material 3](https://m3.material.io/styles/color/static/baseline) color scheme.
+  /// - System colors, if available.
+  ///
+  /// This example demonstrates how to use the high contrast themes with [MaterialApp]:
+  ///
+  /// ```dart
+  /// MaterialApp(
+  ///   theme: ThemeData.light(),
+  ///   darkTheme: ThemeData.dark(),
+  ///   highContrastTheme: ThemeData.highContrastLight(),
+  ///   highContrastDarkTheme: ThemeData.highContrastDark(),
+  ///
+  ///   home: Scaffold(
+  ///     appBar: AppBar(
+  ///       title: const Text('Home'),
+  ///     ),
+  ///   ),
+  /// )
+  /// ```
+  factory ThemeData.highContrastLight({bool useMaterial3 = true}) {
```

Comment:
The use case for this seems a bit limited to me if we are not going to expose seed color. If this is just meant to be a default, we should probably make this a private method in MaterialApp.

Reply:
This is a convenient theme like `ThemeData.light()` and `ThemeData.dark()` which are also limited. Why do you want to make it private?

Reply:
ah I see yeah, I kept thinking they will have option to provide additional color or component theme in that case, I was wrong. This is fine then